packages/typescript/src/transactions/Transaction.ts

```diff
@@ -678,6 +687,37 @@ export class Transaction {
 		return signer.signTransaction(bytes);
 	}
 
+	/**
+	 *  Ensures that:
+	 *  - All objects have been fully resolved to a specific version
+	 *  - All pure inputs have been serialized to bytes
+	 *  - All async thunks have been fully resolved
+	 *  - All transaction intents have been resolved
+	 * 	- The gas payment, budget, and price have been set
+	 *  - The transaction sender has been set
+	 *
+	 *  When true, the transaction will always be built to the same bytes and digest (unless the transaction is mutated)
+	 */
+	isFullyResolved() {
+		if (!this.#data.sender) {
+			return false;
+		}
+
+		if (this.#pendingPromises.size > 0) {
```

Comment:
I'm guessing this is why the data itself can't know if it's fully resolved or not?